src/api/sfc-tooling.md

```diff
@@ -0,0 +1,94 @@
+
+[WebStorm](https://www.jetbrains.com/webstorm/) also provides decent support for Vue SFCs. However, do note as of now its support for `<script setup>` is still [in progress](https://youtrack.jetbrains.com/issue/WEB-49000).
+
+Most other editors have community-created syntax highlighting support for Vue, but lack the same level of code IntelliSense. In the long run, we do hope we can extend the range of editor support by leveraging the [Language Service Protocol](https://microsoft.github.io/language-server-protocol/) as Volar's core logic is implemented as a standard language server.
+
+## Testing Support
+
+- If using Vite, we recommend [Cypress](https://www.cypress.io/) as the test runner for both unit and e2e tests. Unit tests for Vue SFCs can be done with the [Cypress Component Test Runner](https://www.cypress.io/blog/2021/04/06/introducing-the-cypress-component-test-runner/).
+
+- Vue CLI comes with [Jest](https://jestjs.io/) and [Mocha](https://mochajs.org/) integrations.
+
+- If you are manually confugring Jest to work with Vue SFCs, check out [vue-jest](https://github.com/vuejs/vue-jest) which is the official Jest transform for Vue SFCs.
+
+## Custom Blocks Integration
+
+Custom blocks are compiled into imports to the same Vue file with different request queries. It is up to the underlying build tool to handle these import requests.
+
+- If using Vite, a custom Vite plugin should be used to transform matched custom blocks into executable JavaScript. [[Example](https://github.com/vitejs/vite/tree/main/packages/plugin-vue#example-for-transforming-custom-blocks)]
+
+- If using Vue CLI or plain webpack, a webpack loader should be configured to transform the matched blocks. [[Example](https://vue-loader.vuejs.org/guide/custom-blocks.html#custom-blocks)]
+
+## Lower Level Tools
+
+### `@vue/compiler-sfc`
+
+- [Docs](https://github.com/vuejs/vue-next/tree/master/packages/compiler-sfc)
+
+This package is part of the Vue core monorepo and is always published with the same version as the main `vue` package. Typically, it will be listed as a peer dependency of `vue` in a project. To ensure correct behavior, its version should always be kept in-sync with `vue` - i.e. whenever you upgrade the version of `vue`, you should also upgrade `@vue/compiler-sfc` to match it.
+
+The package itself provides lower-level utilities for processing Vue SFCs and is only meant for tooling authors that need to support Vue SFCs in custom tools.
+
+### `@vitejs/plugin-vue`
+
+- [Docs](https://github.com/vitejs/vite/tree/main/packages/plugin-vue)
+
+Official plugin that provides Vue SFC support in Vite.
+
+### `vue-loader`
+
+- [Docs](https://vue-loader.vuejs.org/)
+
+The official loader that provides Vue SFC support in webpack. If you are using Vue CLI, also see [docs on modifying `vue-loader` options in Vue CLI](https://cli.vuejs.org/guide/webpack.html#modifying-options-of-a-loader).
```

Comment:
I think it would be better to include 3rd parties using custom blocks, as it would be more helpful for people who are creating ecosystems for some ideas.

As far as I know, it is the following

- [gridsome](https://gridsome.org/): `<page-query>`
- [vite-plguin-vue-gql](https://github.com/wheatjs/vite-plugin-vue-gql): `<gql>`
- [vue-i18n](https://github.com/intlify/bundle-tools): `<i18n>`

Reply:
I think this should be added to the SFC spec page

Reply:
see a4cf142